File: javascript/script/quicksort.js
function quicksort (arr) {
    if (arr.length < 2) return arr

    let pivot = arr[0]
    let left = arr.slice(1).filter(num => num <= pivot)
    let right = arr.slice(1).filter(num => num > pivot)

    return [...quicksort(left), pivot, ...quicksort(right)]
}

module.exports = quicksort

File: python/script/quicksort.py
def quicksort(arr):
    if len(arr) < 2:
        return arr

    pivot = arr[0]
    left = [i for i in arr[1:] if i <= pivot]
    right = [i for i in arr[1:] if i > pivot]

    return quicksort(left) + [pivot] + quicksort(right)
